File: FinancialPortfolioClient/FinancialPortfolio.Client/src/core/models/system-log/client-log-request.ts
import { LogLevel } from '../../enums/logs/log-level.enum';

export interface ClientLogRequest {
  level: LogLevel;
  category: string;
  method?: string;
  message: string;
  exception?: string | null;
  stackTrace?: string | null;
  pageUrl?: string;
  userAgent?: string;
}

File: FinancialPortfolioServer/FinancialPortfolio.Models/Model/Request/SystemLog/ClientLogRequest.cs
using FinancialPortfolio.Models.Common.Enums;

namespace FinancialPortfolio.Models.Model.Request.SystemLog
{
    public class ClientLogRequest
    {
        public LogLevelType Level { get; set; } = LogLevelType.Error;           // Information, Warning, Error, Critical
        public string Category { get; set; } = "Angular";
        public string Method { get; set; } = string.Empty;
        public string? Message { get; set; } = string.Empty;
        public string? Exception { get; set; }
        public string? StackTrace { get; set; }
        public string? PageUrl { get; set; }
        public string? UserAgent { get; set; }
    }
}
